File: workflow/plugins/go/string/string_replace/string_replace.go
// Package string_replace provides a workflow plugin for replacing strings.
package string_replace

import (
	"strings"
)

// StringReplace implements the NodeExecutor interface for replacing strings.
type StringReplace struct {
	NodeType    string
	Category    string
	Description string
}

// NewStringReplace creates a new StringReplace instance.
func NewStringReplace() *StringReplace {
	return &StringReplace{
		NodeType:    "string.replace",
		Category:    "string",
		Description: "Replace occurrences in a string",
	}
}

// Execute runs the plugin logic.
func (p *StringReplace) Execute(inputs map[string]interface{}, runtime interface{}) map[string]interface{} {
	str, ok := inputs["string"].(string)
	if !ok {
		return map[string]interface{}{"result": "", "error": "string is required"}
	}

	old, _ := inputs["old"].(string)
	new, _ := inputs["new"].(string)

	// Default to replace all (-1)
	count := -1
	if n, ok := inputs["count"].(int); ok {
		count = n
	}

	result := strings.Replace(str, old, new, count)
	return map[string]interface{}{"result": result}
}

File: workflow/plugins/python/string/string_replace/string_replace.py
"""Workflow plugin: replace in string."""

from ...base import NodeExecutor


class StringReplace(NodeExecutor):
    """Replace occurrences in string."""

    node_type = "string.replace"
    category = "string"
    description = "Replace occurrences in string"

    def execute(self, inputs, runtime=None):
        value = str(inputs.get("value", inputs.get("text", "")))
        old = inputs.get("old", inputs.get("search", ""))
        new = inputs.get("new", inputs.get("replacement", ""))
        count = inputs.get("count", -1)
        return {"result": value.replace(old, new, count)}
